superset-frontend/src/components/OmniContainer/getDashboards.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { t, SupersetClient } from '@superset-ui/core';
+import to from 'await-to-js';
+
+interface DashboardItem {
+  changed_by_name: string;
+  changed_on: string;
+  creator: string;
+  dashboard_link: string;
+  dashboard_title: string;
+  id: number;
+  modified: string;
+  url: string;
+}
+
+interface Dashboards extends DashboardItem {
+  title: string;
+}
+
+export const getDashboards = async (
+  query: string,
+): Promise<(Dashboards | { title: string })[]> => {
+  // todo: Build a dedicated endpoint for dashboard searching
+  // i.e. superset/v1/api/dashboards?q=${query}
+  const [error, response] = await to(
+    SupersetClient.get({
+      endpoint: `/dashboardasync/api/read?_oc_DashboardModelViewAsync=changed_on&_od_DashboardModelViewAsync=desc&_flt_2_dashboard_title=${query}`,
+    }),
+  );
+  if (error) {
+    return [{ title: t('An error occurred while fetching dashboards') }];
+  }
```

Comment:
```suggestion
  let response;
  try {
    response = await SupersetClient.get({
      endpoint: `/dashboardasync/api/read?_oc_DashboardModelViewAsync=changed_on&_od_DashboardModelViewAsync=desc&_flt_2_dashboard_title=${query}`,
    });
  } catch (error) {
    return [{ title: t('An error occurred while fetching dashboards') }];
  }
```

Reply:
@ktmud  I know that try / catch produces the same result, but I wanted to show some points. I really like to use `await to` because the code is much more readable and easier to understand.
```ts
const [err, resp] = asyncFunc();
if(err)// do something
// do something

vs

try{
const resp = asyncFunc();
// do something
}catch(err){
// do something
}

```

The first option looks much more like the node's  error first pattern.
It also makes requests a lot easier. For example, if same request returns an error (status 400) and some useful information in the body (error messages), with `await to` it is very simple to capture.

In general, I try to avoid `try/catch` as they break the error first pattern and makes the code a little more complex to read. Not that `try/catch` is very hard, it is because the `await to` is much easier.
If you still disagree, I can remove it, but I think it helps to decrease code complexity.

Reply:
Let's try not to add a new dependency. Try/catch is universal and easy enough, IMO. It seems all this library does it transforming error handling into another style which not all developers are familiar with. It takes some cognitive load understand what this library does, even it's something dead simple---people would wonder, sure, this seems to handle errors from a promise, but does it do anything extra?

In the future, all async API requests and error handling may be abstracted away by the [`useResource`](https://github.com/apache/superset/pull/13218) hook @suddjian is working on so this is probably even more unnecessary.

Even right now, [`makeApi`](https://github.com/apache-superset/superset-ui/blob/482da217938bcbd70db8b79117abacf00f8037e2/packages/superset-ui-core/test/query/api/v1/makeApi.test.ts#L170-L187) has already made it easy to handle response errors (it ensures erroring HTTP status code and errors in response body always throw and capturable by `try/catch`).